### DISCUSSION TITLE
Is there any error handling when loading assets?

### DISCUSSION THREAD

**Discussion by aram88888** [pixijs/pixijs#9629] [General]
Hi. How can I detect which file failed to load? Something from your examples.
 import { Assets } from 'pixi.js';

 Assets.addBundle('animals', {
     bunny: 'missingFile.png',
     chicken: 'chicken.png',
     thumper: 'thumper.png',
 });

Assets.loadBundle('animals').then(t=>{
}).catch(error=>{
// How can I know the missing file name?
});

**Comment by SuperSodaSea** [comment]
Right now it looks like we can't do that, we simply `throw new Error`.

https://github.com/pixijs/pixijs/blob/0e875b5acb32e163a8a79a5c9d19d631689591ef/packages/assets/src/loader/Loader.ts#L207

If you want PixiJS to throw a detailed error class when loading a file fails (change this line into something like `new AssetError(url, e)`, so that you can access `e.url` & `e.cause`), you can make a [Feature Request](https://github.com/pixijs/pixijs/issues/new?assignees=&labels=Feature+Request&projects=&template=feature_request.yml&title=Feature+Request%3A+).
